RIOT-OS/RIOT
# tests/posix_semaphore: better guarantee sequence of output lines

### Contribution description

As suggested by @kaspar030 in #14281 the THREAD_CREATE_WOUT_YIELD flag was removed.
Now the sequence of output lines is identical in two cases: stdio via UART,
and stdio via CDC ACM (which needs a thread for that).

The expects have been changed to match the new situation, which is what
was already given in the CDC ACM case.

### Testing procedure

Run `tests/posix_semaphore` with `compile_and_test_for_board`. Choose a board which uses CDC ACM as stdio. That means there are three threads active when the test starts. The test should succeed.

Run `tests/posix_semaphore` with `compile_and_test_for_board`. Choose a board which uses standard UART for stdio (no CDC ACM). That means there are two threads active when the test starts. The test should succeed.


### Issues/PRs references

Fixes #14281
